### ISSUE TITLE
Regression on Android build

### ISSUE THREAD

**Issue by Gae24** [servo/servo#32161]
**Describe the bug:**
After the latest changes to the font system the android app doesn't work anymore.

```thread 'FontCache' panicked at components/gfx/platform/freetype/android/font_list.rs:41:14:
Couldn't open font file!: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:645:13:
Failed to receive a response from live font cache
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:613:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Style#1' panicked at components/gfx/font_cache_thread.rs:613:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Style#2' panicked at components/gfx/font_cache_thread.rs:639:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Style#1' panicked at components/gfx/font_cache_thread.rs:613:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:639:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:639:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:639:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Style#1' panicked at components/gfx/font_cache_thread.rs:613:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Style#2' panicked at components/gfx/font_cache_thread.rs:639:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Script(1,1)' panicked at components/gfx/font_cache_thread.rs:613:14:
failed to send message to font cache thread: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread '<unnamed>' panicked at components/gfx/font_cache_thread.rs:371:59:
called `Result::unwrap()` on an `Err` value: Io(Custom { kind: BrokenPipe, error: "crossbeam-channel receiver closed" })
thread 'Constellation' panicked at components/constellation/constellation.rs:1231:21:
Unexpected script channel panic in constellation: RecvError
thread 'StorageManager' panicked at components/net/storage_thread.rs:61:36:
called `Result::unwrap()` on an `Err` value: Disconnected
thread 'ResourceManager' panicked at components/net/resource_thread.rs:229:45:
called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Other crossbeam-channel error" } ```

**Comment by mukilan** [comment]
I'm not able to observe this issue on my Android device, with a build from latest master.

Could you please provide more information about the issue such as:
1. Whether this is on an emulator or a real device?
2. Version of Android running on the device/emulator.
3. Architecture of the device/emulator (arm or intel, 32 bit or 64 bit).
4. Steps to reproduce the issue: Is this intermittent? or consistently reproducible? Did you have to load a different page than the default servo.org page?

**Comment by Gae24** [comment]
I'm testing on a Samsung A52s with the armv7 build, happens always the servo page is not rendered correctly [https://ibb.co/gdhsfBM](url)

**Comment by mukilan** [comment]
@Gae24 In theory, the changes in #32298 should have fixed this issue, but I couldn't validate reproduce this on my device so it would be great if you could confirm if the issue is fixed on your device after #32298.

**Comment by mukilan** [comment]
Thanks for confirming!
